### PR TITLE
EXSWHTEC-110-Implement Texture Management GET api tests

### PR DESCRIPTION
**Pull request by marko-veniger**
Implemented unit tests for the following APIs:
- hipGetChannelDesc
- hipGetTextureObjectResourceDesc
- hipGetTextureObjectResourceViewDesc
- hipGetTextureObjectTextureDesc
- hipTexObjectGetResourceDesc
- hipTexObjectGetResourceViewDesc
- hipTexObjectGetTextureDesc
- hipDestroyTextureObject
- hipTexObjectDestroy

### CODE REVIEW

File: catch/unit/texture/hipTexObjectTests.cc [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  memset(&check_desc, 0, sizeof(check_desc));
+
+  HIP_CHECK(hipTexObjectGetResourceViewDesc(&check_desc, tex_obj_wrapper.texture_object));
+
+  REQUIRE(check_desc.format == tex_obj_wrapper.res_view_desc.format);
+  REQUIRE(check_desc.width == tex_obj_wrapper.res_view_desc.width);
+}
+#endif
+
+#if HT_AMD
```
Comment (chrispaquot) [inline catch/unit/texture/hipTexObjectTests.cc]: can you just remove the test from json for nv?

File: catch/unit/texture/hipTextureObjectTests.cc [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  SECTION("texture is invalid") {
+    HIP_CHECK_ERROR(
+        hipGetTextureObjectResourceDesc(&check_desc, static_cast<hipTextureObject_t>(0)),
+        hipErrorInvalidValue);
+  }
+}
+
+/* hipGetTextureObjectResourceViewDesc tests */
+
+#if HT_AMD
```
Comment (chrispaquot) [inline catch/unit/texture/hipTextureObjectTests.cc:131]: same

File: catch/unit/texture/hipTextureObjectTests.cc [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  memset(&check_desc, 0, sizeof(check_desc));
+
+  HIP_CHECK(hipGetTextureObjectResourceViewDesc(&check_desc, tex_obj_wrapper.texture_object));
+
+  REQUIRE(check_desc.format == tex_obj_wrapper.res_vew_desc.format);
+  REQUIRE(check_desc.width == tex_obj_wrapper.res_vew_desc.width);
+}
+#endif
+
+#if HT_AMD
```
Comment (chrispaquot) [inline catch/unit/texture/hipTextureObjectTests.cc:147]: again

**Review by chrispaquot — CHANGES_REQUESTED** [review]
Remove the test cases with HT_AMD to json instead.

File: catch/unit/texture/hipTexObjectTests.cc (hunk 1 above)
Comment (nives-vukovic) [reply catch/unit/texture/hipTexObjectTests.cc]: These tests have been modified to work on Nvidia and are enabled now.

File: catch/unit/texture/hipTextureObjectTests.cc (hunk 2 above)
Comment (nives-vukovic) [reply catch/unit/texture/hipTextureObjectTests.cc:131]: @chrispaquot   Unfortunately, it is not possible as hipGetTextureObjectResourceViewDesc does not compile on Nvidia.

File: catch/unit/texture/hipTextureObjectTests.cc (hunk 3 above)
Comment (nives-vukovic) [reply catch/unit/texture/hipTextureObjectTests.cc:147]: @chrispaquot  Same as above. Same also for tests that call hipGetTextureObjectTextureDesc.

File: catch/unit/texture/hipTextureObjectTests.cc [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+                    hipErrorInvalidValue);
+  }
+
+  SECTION("texture is invalid") {
+    HIP_CHECK_ERROR(
+        hipGetTextureObjectResourceViewDesc(&check_desc, static_cast<hipTextureObject_t>(0)),
+        hipErrorInvalidValue);
+  }
+
+  HipTest::HIP_SKIP_TEST("Skipping on NVIDIA platform");
```
Comment (chrispaquot) [inline catch/unit/texture/hipTextureObjectTests.cc]: you don't need this here then?

Comment (nives-vukovic) [reply catch/unit/texture/hipTextureObjectTests.cc]: @chrispaquot Removed skipping of nvidia test that is disabled.

Comment (nives-vukovic) [reply catch/unit/texture/hipTextureObjectTests.cc]: @rakesroy It seems all changes made in this commit have been already merged as part of https://github.com/ROCm/hip-tests/pull/92 . When you verify this, the PR can be closed.

**Comment by rakesroy** [comment]
All changes made in this commit have been already merged as part of https://github.com/ROCm/hip-tests/pull/92.

**Pull request closed without merging**